pkg/manager/namespace.go

```diff
@@ -27,13 +27,18 @@ import (
 )
 
 const (
-	WatchNamespaceEnvVar = "WATCH_NAMESPACE"
+	WatchNamespaceEnvVar = "WATCH_NAMESPACE" // We need to address WATCH_NAMESPACE to the new layout.
+	// It currently was not added
 )
 
 func ConfigureWatchNamespaces(options *manager.Options, log logr.Logger) {
 	namespaces := lookupEnv()
 	if len(namespaces) != 0 {
 		log.Info("watching namespaces", "namespaces", namespaces)
+		// Currently: cache.MultiNamespacedCacheBuilder(strings.Split(namespace, ","))
+		// because for SDK OLM features and command line arg to work with is an string
+		// However, I agree that use array here is better we just need to first
+		// to do the required checks and fixes in SDK.
 		if len(namespaces) > 1 {
```

Comment:
TODO ^WDYT?

Reply:
See that if namespaces = "value1,value2"
Them, the if will NOT return true for > 1 

So, my point is
- I agree that it can be ana array
- But it is a breaking change since currently the flags --namespace and --operator-namespace expects a string and not an array. See: https://github.com/operator-framework/operator-sdk/search?q=namespace&unscoped_q=namespace

Reply:
`namespaces` is a slice returned by `lookupEnv()`. It can't be a string. `lookupEnv()` looks up the `WATCH_NAMESPACE` environment variable value (which can be a comma-separated string), and splits it into a slice.

Step through the logic of `ConfigureWatchNamespaces`, `lookupEnv`, and `splitNamespaces` one more time, and check out the [tests](https://github.com/joelanford/helm-operator/blob/563f188848e0df95bbd58440ce0ed41d28572d64/pkg/manager/namespace_test.go#L40). I think you'll come to the conclusion that this function properly handles WATCH_NAMESPACE being a comma-separated string.

BTW, `--namespace` and `--operator-namespace` are not flags of the helm-operator. To see the flags supported by the current helm operator, run `operator-sdk exec-entrypoint helm -h`.

Reply:
You are right about 
- the --namepsace and operator-namespace. 
- Regards the WATCH_NASMACE see that I  agree that it is better as an array, however currently we have for helm: 

$ operator-sdk run local  --help
```
     --watch-namespace string      The namespace where the operator watches for changes. Set "" for AllNamespaces, set "ns1,ns2" for MultiNamespace
 ```

The most important point to beware of which in my POV we need to check:

- If we gen a CSV of the operator and set up it was multi-namespace the values from OperatorGroup will be set in the WATCH_NAMESPACE as a string "ns1,ns2".  So, we need to ensure that it will NOT be a string. 

Related to : https://github.com/operator-framework/operator-sdk/blob/d7a0ecc70018dfae8661fd52e8656d954809ac8f/internal/generate/clusterserviceversion/clusterserviceversion_updaters.go#L126 and https://github.com/operator-framework/operator-sdk/blob/master/internal/generate/olm-catalog/csv.go#L392-L414

- Check if we have any OLM subcommand which allows pass installMode as multi-namespace and it will be a string: (I do not think that it still possible, but shows a point to check as well)

```
Flag --install-mode has been deprecated, use this flag with 'run packagemanifests' instead
FATA[0000] Failed to run operator using OLM: validation error: installMode string "--help" is malformatted, must be: InstallModeType[=ns1,ns2[, ...]] 

```
More info: https://github.com/operator-framework/operator-sdk/issues/3001 and https://github.com/operator-framework/operator-sdk/issues/2361#issuecomment-581073076

Reply:
@camilamacedo86 Again, this code handles and expects `WATCH_NAMESPACE` as a comma-separated string. It converts it from a string to a slice because `cache.MultiNamespacedCacheBuilder()` needs it to be a slice.